core/trino-main/src/main/java/io/trino/execution/Input.java

```diff
@@ -32,6 +32,7 @@
 public final class Input
 {
     private final String catalogName;
+    private final String catalogVersion;
```

Comment:
Is there anyway we can use the Object form of `CatalogVersion` here instead of string?

Reply:
I changed the type of the field to `Object` instead of `String` / `CatalogHandle.CatalogVersion`.
The only constraint is now on the constructor to have `CatalogHandle.CatalogVersion` as a parameter so that it knows during the deserialization that the value 

```
"catalogVersion": "default"
```

is to be converted back to a `CatalogHandle.CatalogVersion` instead of `String` like it would normally happen without this extra information.